core/src/main/java/gyro/core/resource/DiffableType.java

```diff
@@ -149,10 +150,11 @@ public DiffableField getField(String name) {
         return null;
     }
 
-    public D newExternal(RootScope root, Object id) {
+    public D newExternal(RootScope root, Object id, String credentials) {
```

Comment:
I'd recommend something like the following rather than adding an argument that can be `null`.

```suggestion
    public D newExternalWithCredentials(RootScope root, Object id) {
        D diffable = newExternal(root, id);
        diffable.scope.getSettings(CredentialsSettings.class).setUseCredentials(credentials);
        
        return diffable;
    }
```

Reply:
Updated based on suggestio